f5lbaasdriver/v2/bigip/service_builder.py

```diff
@@ -201,6 +208,18 @@ def _get_extended_member(self, context, member):
 
         return (member_dict, subnet, network)
 
+    @log_helpers.log_method_call
+    def _get_extended_qos(self, context, loadbalancer):
+        try:
+            sess = keystone.get_session()
+            client_sess = client.Client(session=sess)
+            project = client_sess.projects.get(project=loadbalancer.tenant_id)
+            return project.qos
```

Comment:
What will project.qos value look like?

Reply:
It is a string like "high" "medium" "low".

And these values come from the bigip side's predefined qos profile's name.

Reply:
Is the tag name 'qos' in keystone project? Maybe a minor suggestion is to name it a little more precisel (eg. instance qos_bandwidth). VS bwc name in L7policy is 'vip_bwc_policy'.

Reply:
Yes, we could change the name if needed. Actually this tag is created by ourselves. I will make the change accordingly.

Thanks.